### DISCUSSION TITLE
Same libnative.so when included in different wasm apps Memory is shared for it?

### DISCUSSION THREAD

**Discussion by shashankprashar98** [bytecodealliance/wasm-micro-runtime#3751] [Q&A]
Hi i wanted to know if I'm including the libnative.so (my native lib) while executing my .wasm apps, then if I'm including the same libnative.so say : 
NATIVE_LIB="./libnative.so"
$IWASM_EXEC --native-lib=$NATIVE_LIB $WASM_FILE

then if i do this for different set of .wasm files, will .wasm apps share the same libnative.so (text segment ? or data segment)
I tried using pmap on different .wasm processes that are running, i can see virtual addresses for libnative.so are coming up different but what about the physical address and physical memory utilization ?

If  we are saying .wasm is running in sandbox like environment, will the libnative.so be replicated in physical memory or will still be shared as it normally does ?

**Comment by TianlongLiang** [comment]
IIUC, you are running multiple processes, for example:
```bash
NATIVE_LIB="./libnative.so"
# one process
$IWASM_EXEC --native-lib=$NATIVE_LIB $WASM_FILE
# another 
$IWASM_EXEC --native-lib=$NATIVE_LIB $WASM_FILE_2
```
The actual implementation of runtime is to use `dlopen` to load the shared library, so it will still be shared as it normally does
